packages/browser/__tests__/Session.spec.ts

```diff
@@ -139,12 +140,26 @@ describe("Session", () => {
       const clientAuthentication = mockClientAuthentication();
       const clientAuthnFetch = jest.spyOn(clientAuthentication, "fetch");
       const mySession = new Session({ clientAuthentication });
+      mySession.info.isLoggedIn = true;
       const objectWithFetch = {
         fetch: mySession.fetch,
       };
       await objectWithFetch.fetch("https://some.url");
       expect(clientAuthnFetch).toHaveBeenCalled();
     });
+
+    // Prevents the bug when clientAuthentication.fetch throws
+    // ""'fetch' called on an object that does not implement interface Window"
+    // when unauthenticated.
+    it("does not call authenticated fetch if logged out", async () => {
+      window.fetch = jest.fn();
+      const clientAuthentication = mockClientAuthentication();
+      const clientAuthnFetch = jest.spyOn(clientAuthentication, "fetch");
+      const mySession = new Session({ clientAuthentication });
+      await mySession.fetch("https://some.url/");
+      expect(clientAuthnFetch).not.toHaveBeenCalled();
+      expect(window.fetch).toHaveBeenCalled();
```

Comment:
You explicitly want to check here whether `window.fetch` has been called with `window` as `this` pointer.

Reply:
Excellent, thanks ! Done in e646adbe43aa0575c16b0f1c54f81830a59ae993